src/JobPanel/backend/async_repeater.py

```diff
@@ -330,28 +392,82 @@ class AsyncRepeater():
     and then propagating back the requests. Repeater do not process requests itself. Only in case of error
     it sends the error answer itself.
 
+    TODO:
+    listen port vzdy ziskat ze systemu (volani s listen_port=0)
+
+
+
     """
-    def __init__(self, repeater_address, listen_port):
+    def __init__(self, repeater_address, listen_port, parent_repeater_address=None):
         """
         :param repeater_address: Repeater id given be parent repeater in the tree.
         :param listen_port: None means no Server, 0 - get by kernel
 
         """
         self.repeater_address = repeater_address
+        self.parent_repeater_address = parent_repeater_address
         self.max_client_id=0
         self.clients = {}
         """ Dict of clients. Keys client_id. """
+        self._starter_client_thread = None
+        self._starter_client_attempting = False
         if (listen_port is None):
             self._server_dispatcher = None
             self.listen_port = None
         else:
             self._server = Server(repeater_address, listen_port, self.clients)
             self.listen_port = self._server.address[1]
             self._server_dispatcher = self._server.get_dispatcher()
+
+            if self.parent_repeater_address is not None:
+                self._starter_client_attempting = True
+                self._starter_client_thread = threading.Thread(target=self._starter_client_run)
```

Comment:
Why we need separate thread, just do it in async_dispatcher loop, in order to make the implementation uniform. Moreover we want a confirmation from the parent repeater. This i'snt done on both sides either.